### DISCUSSION TITLE
Always show border when it's not selected

### DISCUSSION THREAD

**Discussion by elpmisu** [fabricjs/fabric.js#6894] [Q&A]
I would like to use Text  border ,and keep  show when it's not selected
How can I do this

## Version
3.6.6

**Comment by melchiar** [comment]
@QiZZ121 can you please clarify what this would look like?

**Reply by elpmisu** [reply]
For exapmle， 

```
var canvas =  new fabric.Canvas('c');

  var text = new fabric.Text("Test", {
        top: 100, 
        left: 100, 
        fill: 'red', 
        fontSize: '14', 
        selectable: true, 
        fontFamily: 'Microsoft Yahei', 
        visible: !!item.showString,
        hasControls: false,
        hasBorders: true,
        borderColor: 'red',
        objectCaching: false 
      });
      canvas.add(text);
```
when i  use  hasBorders and borderColor to change border，
 I want the border to be displayed all the time instead of being displayed after being selected
What should I do to achieve it?
Thanks!

**Reply by melchiar** [reply]
If you mean showing both the borders and controls of all objects selected or not, this is not supported since only selected objects show controls. If you're okay with just the borders see the demo below for how to use the `after:render` event to run draw commands on all objects in your canvas.

http://fabricjs.com/bounding-rectangle

**Comment by asturur** [comment]
This would be a different border right?

The library has some drawing code to represent a selection state.

You want to draw a border around the objects, but not having them selected.
Reusing the selection state does not seem a great idea. 

The example linked from @melchiar makes sense.

**Comment by logone72** [comment]
If your goal is to keep a specific object selected (rendering borders and controls), I believe this is a good workaround.

```js
canvas.on('selection:cleared', function (e) {
      canvas.setActiveObject(obj);
      canvas.renderAll();
});
```
